pkg/promutil/migrate.go

```diff
@@ -64,63 +63,89 @@ func BuildMetrics(results []model.CloudwatchMetricResult, labelsSnakeCase bool,
 	for _, result := range results {
 		contextLabels := contextToLabels(result.Context, labelsSnakeCase, logger)
 		for _, metric := range result.Data {
-			for _, statistic := range metric.Statistics {
-				var includeTimestamp bool
-				if metric.AddCloudwatchTimestamp != nil {
-					includeTimestamp = *metric.AddCloudwatchTimestamp
-				}
-				exportedDatapoint, timestamp, err := getDatapoint(metric, statistic)
+			// This should not be possible but check just in case
+			if metric.GetMetricStatisticsResult == nil && metric.GetMetricDataResult == nil {
+				logger.Warn("Attempted to migrate metric with no result", "namespace", metric.Namespace, "metric_name", metric.MetricName, "resource_name", metric.ResourceName)
+			}
+
+			for _, statistic := range statisticsInCloudwatchData(metric) {
+				dataPoint, ts, err := getDatapoint(metric, statistic)
 				if err != nil {
 					return nil, nil, err
 				}
-				if exportedDatapoint == nil && (metric.AddCloudwatchTimestamp == nil || !*metric.AddCloudwatchTimestamp) {
-					exportedDatapoint = aws.Float64(math.NaN())
-					includeTimestamp = false
-					if *metric.NilToZero {
-						exportedDatapoint = aws.Float64(0)
-					}
+				var exportedDatapoint float64
+				if dataPoint == nil && metric.MetricMigrationParams.AddCloudwatchTimestamp {
+					// If we did not get a datapoint then the is default making it unusable in the exported metric
```

Comment:
I don't understand this line. Is there a typo near "then the is default making it unusable"? Should it say something like "there is no default timestamp" or "then the timestamp is also missing"?

Reply:
100% a typo, I'll reword it